tests/functional/testOnboarding.js

```diff
@@ -277,4 +280,69 @@ describe('Onboarding', function() {
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.PRIVACY_NEXT_BUTTON.visible());
+    await vpn.waitForQuery(queries.screenOnboarding.STEP_NAV_STACK_VIEW.ready());
+    assert.equal(await vpn.getSetting('gleanEnabled'), true);
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.DEVICES_NEXT_BUTTON.visible());
+    await vpn.waitForQuery(queries.screenOnboarding.STEP_NAV_STACK_VIEW.ready());
+    assert.equal(await vpn.getSetting('gleanEnabled'), true);
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.START_NEXT_BUTTON.visible());
+    await vpn.waitForQuery(queries.screenHome.SCREEN.visible());
+    assert.equal(await vpn.getSetting('onboardingCompleted'), true);
+
+    //Now that onboarding is complete, make sure the gleanEnabled setting is now disabled
+    assert.equal(await vpn.getSetting('gleanEnabled'), false);
+
+    //Now repeat the process, but we will finish onboarding with the data collection checkbox being checked
+
+    //Reset settings, logout, and authenticate
+    await vpn.setSetting('onboardingCompleted', false)
+    await vpn.setSetting('gleanEnabled', true);
+    await vpn.logout();
+    await vpn.authenticateInApp();
+
+    //Ensure that the data collection checkbox and internal setting are decoupled
+    //We will finish onboarding with the data collection checkbox being checked
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.DATA_CHECKBOX, 'isChecked'), 'false');
+    assert.equal(await vpn.getSetting('gleanEnabled'), true);
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.DATA_CHECKBOX.visible());
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.DATA_CHECKBOX, 'isChecked'), 'true');
+    assert.equal(await vpn.getSetting('gleanEnabled'), true);
+
+    //Complete onboarding, and check that the gleanEnabled setting does not change between steps
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.DATA_NEXT_BUTTON.visible());
+    await vpn.waitForQuery(queries.screenOnboarding.STEP_NAV_STACK_VIEW.ready());
+    assert.equal(await vpn.getSetting('gleanEnabled'), true);
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.PRIVACY_NEXT_BUTTON.visible());
+    await vpn.waitForQuery(queries.screenOnboarding.STEP_NAV_STACK_VIEW.ready());
+    assert.equal(await vpn.getSetting('gleanEnabled'), true);
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.DEVICES_NEXT_BUTTON.visible());
+    await vpn.waitForQuery(queries.screenOnboarding.STEP_NAV_STACK_VIEW.ready());
+    assert.equal(await vpn.getSetting('gleanEnabled'), true);
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.START_NEXT_BUTTON.visible());
+    await vpn.waitForQuery(queries.screenHome.SCREEN.visible());
+    assert.equal(await vpn.getSetting('onboardingCompleted'), true);
+
+    //Now that onboarding is complete, make sure the gleanEnabled setting is still enabled
+    assert.equal(await vpn.getSetting('gleanEnabled'), true);
+  });
```

Comment:
Please add a test case for when the application is closed and reopened during onboarding.